### PR TITLE
feat(inputs.sql): Add FlightSQL support

### PR DESCRIPTION
**Pull request by srebhan**
- [ ] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

This adds support for FlightSQL databases.

### CODE REVIEW

**Comment by srebhan** [comment]
Build fails due to upstream issue (reported as https://github.com/apache/arrow/issues/35133).

**Comment by srebhan** [comment]
This PR https://github.com/apache/arrow/pull/35137 is required to go into arrow first so we can get rid of the replace statement.

File: plugins/inputs/sql/sql.go [hunk 1]
```diff
@@ -392,7 +392,8 @@ func (s *SQL) Start(_ telegraf.Accumulator) error {
 		stmt, err := s.db.PrepareContext(ctx, q.Query)
 		cancel()
 		if err != nil {
-			return fmt.Errorf("preparing query %q failed: %w", q.Query, err)
+			s.Log.Warnf("preparing query %q failed: %w; falling back to unprepared query", q.Query, err)
+			continue
```
Comment (powersj) [inline plugins/inputs/sql/sql.go:396]: Can you add a comment explaining, why the change as this affects all drivers

**Review by powersj — COMMENTED** [review]
I am weary of changing errors to warnings with the wide support of drivers this has. Could you provide some more context in line?

File: plugins/inputs/sql/sql.go [hunk 2]
```diff
@@ -450,14 +451,23 @@ func init() {
...
+	// Execute the query either prepared or unprepared
+	var rows *dbsql.Rows
+	if q.statement != nil {
+		var err error
+		rows, err = q.statement.QueryContext(ctx)
+		if err != nil {
+			return err
+		}
+	} else {
+		fmt.Println("Falling back to non-prepared query.")
```
[the remark this answers is not in the document]
Comment (srebhan) [reply plugins/inputs/sql/sql.go:463]: Sorry, this is a left over we probably should remove this altogether and log it only once in `Init()`...

**Comment by srebhan** [comment]
Welcome to dependency hell... If I use the arrow version supporting FlightSQL without prepare statement, `github.com/apache/arrow/go/v12@42527e153286b462ae57e82f876a2443816c82d1` (which is the latest v12 I can find in git), go downgrades `bigquery` to a v11 arrow and builds fail for 32-bit platforms.
When using bigquery v1.51.1 which works on 32-bit platforms it comes with `arrow v12.0.0` which go thinks is _newer_ than the above snapshot. However, the tagged version does not have the "FlightSQL without prepare statement" patch :facepalm:...

**Review by powersj — APPROVED** [review]
